tests/e2e/contracts/log_indexer.py

```
@@ -0,0 +1,68 @@
+        result = list(self.vector_store.knn(emb, 1))
+        if len(result) == 0:
+            return None
+        result = result[0]
+        return {
+            "vector": list(str(x) for x in result.key),
+            "similarity": str(1 - result.distance),
+            "id": result.value.log_id,
+            "text": result.value.text,
+        }
+
+    @gl.public.write
```

Ruff (0.11.9)

`gl` may be undefined, or defined from star imports

(F405)